src/plugins/hotkeyPlugin.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * ショートカットキーを管理するプラグイン。
+ *
+ * HotkeyAction: 実行する処理の名前とコールバックのペア
+ * Combination: ショートカットキーを文字列で表したもの
+ * HotkeySetting: ユーザーが設定できるもの。ActionとCobinationのペア
+ */
+
+/*
+ * 用語メモ：
+ * action: 何をするか
+ * combination: 設定で使う、キーの文字列表記
+ * binding: hotkeys-js に登録したコールバック
+ * bindingKey: hotkeys-js で使う、キーの文字列表記
+ */
+import { Plugin, inject } from "vue";
+import hotkeys from "hotkeys-js";
+import { HotkeyActionNameType, HotkeySettingType } from "@/type/preload";
+
+const hotkeyManagerKey = "hotkeyManager";
+export const useHotkeyManager = () => {
+  const hotkeyManager = inject<HotkeyManager>(hotkeyManagerKey);
+  if (!hotkeyManager) {
+    throw new Error("hotkeyManager not found");
+  }
+  return hotkeyManager;
+};
+
+const log = (message: string, ...args: unknown[]) => {
+  console.log(`[HotkeyManager] ${message}`, ...args);
+};
+
+/**
+ * ショートカットキーの処理を登録するための型。
+ */
+type HotkeyAction = {
+  /** どちらのエディタで有効か */
+  editor: "talk" | "song";
+  /** テキストボックス内で有効か。デフォルトはfalse。 */
+  enableInTextbox?: boolean;
+  /** 名前。 */
+  name: HotkeyActionNameType;
+  /** ショートカットキーが押されたときの処理。 */
+  callback: (e: KeyboardEvent) => void;
+};
+type HotkeyActionId = `${"talk" | "song"}:${HotkeyActionNameType}`;
+
+const actionToId = (entry: HotkeyAction): HotkeyActionId =>
+  `${entry.editor}:${entry.name}`;
+
+/**
+ * ショートカットキーの管理を行うクラス。
+ */
+export class HotkeyManager {
+  private actions: HotkeyAction[] = [];
+  private settings: HotkeySettingType[] = [];
```

Comment:
（ちょっと細かい提案ですが）
これは「全部存在するか１つも存在しないか」だと思うので、それがわかるよう型を明示するのはどうでしょう？
```suggestion
  private settings: HotkeySettingType[] | undefined = undefined;
```